src/weekly/mod.rs

```diff
@@ -51,6 +51,71 @@ impl WeeklyCalendar {
         self.windows.query_point(timepoint).count() > 0
     }
 
+    /// Return the duration remaining till the next window containing the given datetime.
+    ///
+    /// This returns `None` if no windows are reachable.
+    pub fn remaining_to_datetime(&self, datetime: &DateTime<Utc>) -> Option<chrono::Duration> {
+        if self.is_empty() {
+            return None;
+        }
+
+        // Already in a window, zero minutes.
+        if self.contains_datetime(&datetime) {
+            return Some(chrono::Duration::zero());
+        }
+
+        let timepoint = utils::datetime_as_weekly_minute(datetime);
+        // Next window is this week, just subtract remaining minutes.
+        if let Some(next) = self
+            .windows
+            .iter_sorted()
+            .find(|x| x.range.start >= timepoint)
```

Comment:
This logic confuses me; why are we looking at all windows versus just the next one?

Reply:
I'm not sure I get your question. This traverses the windows tree and stops at the [first one](https://doc.rust-lang.org/nightly/core/iter/trait.Iterator.html#method.find) which occurs after the given datetime, if any.
As such window may not exist, the lines afterwards take care of wrapping to the next week.

Reply:
I didn't look in depth at the data structures here but wouldn't the window tree in sorted order give us the first window we're looking for, and we just need to determine whether or not it's this week?

Reply:
I see, I think you are suggesting the `find_next` to be tree-traversing (log) instead of iter-traversing (linear). The library does not expose that primitive right now. I'll keep this code for now and see if I can augment the library for later.